lib/langDialog/lang_dialog.dart

```diff
@@ -0,0 +1,189 @@
+import 'dart:core';
+
+import 'package:auto_size_text/auto_size_text.dart';
+import 'package:flutter/material.dart';
+import 'package:scroll_to_index/scroll_to_index.dart';
+
+class LangDialog extends StatefulWidget {
+  const LangDialog({Key? key}) : super(key: key);
+
+  @override
+  _LangDialogState createState() => _LangDialogState();
+}
+
+class _LangDialogState extends State<LangDialog> {
+  final AutoScrollController controller = AutoScrollController();
+
+  final List<String> _list = [
+    'Kannada',
+    'English',
+    'Hindi',
+    'Telugu',
+    'Tamil',
+    'Marati',
+    'Oriya',
+    'Gujarati',
+    'Panjabi',
+    'French',
+    'Spanish',
+    'Portuguese',
+    'Vietnamese',
+    'Rajastani',
+  ];
+
+  int lastSelectedIndex = 12;
+
+  Future _scrollToIndex(int index) async {
+    await controller.scrollToIndex(index,
+        duration: const Duration(milliseconds: 1500),
+        preferPosition: AutoScrollPosition.begin);
+  }
+
+  @override
+  void initState() {
+    // TODO: implement initState
+    super.initState();
+    _scrollToIndex(lastSelectedIndex);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Dialog(
```

Comment:
Please set min font where you use autosize text. min font should be 8 less that the actual size and also set max lines

Reply:
changes are done